app/src/org/commcare/views/widgets/RecordingFragment.java

```diff
@@ -469,16 +469,35 @@ private void unregisterAudioRecordingConfigurationChangeCallback() {
         }
     }
 
-    @RequiresApi(api = Build.VERSION_CODES.Q)
     private boolean hasRecordingGoneSilent(List<AudioRecordingConfiguration> configs) {
-        if (recorder == null || recorder.getActiveRecordingConfiguration() == null) {
+        if (recorder == null) {
             return false;
         }
 
-        Optional<AudioRecordingConfiguration> currentAudioConfig =
-                configs.stream().filter(config -> config.getClientAudioSessionId()
-                                == recorder.getActiveRecordingConfiguration().getClientAudioSessionId())
-                        .findAny();
-        return currentAudioConfig.isPresent() ? currentAudioConfig.get().isClientSilenced() : false;
+        if (android.os.Build.VERSION.SDK_INT >= android.os.Build.VERSION_CODES.Q) {
+            if (recorder.getActiveRecordingConfiguration() == null) {
+                return false;
+            }
+
+            Optional<AudioRecordingConfiguration> currentAudioConfig = configs.stream().filter(config ->
+                            config.getClientAudioSessionId() == recorder.getActiveRecordingConfiguration()
+                                    .getClientAudioSessionId())
+                    .findAny();
+            return currentAudioConfig.isPresent() ? currentAudioConfig.get().isClientSilenced() : false;
+        } else {
+            // this loop is to get at least 3 zero amplitude readings to confirm that the microphone has gone
+            // silent
+            for (int i = 0; i < 3; i++) {
+                if (recorder == null || recorder.getMaxAmplitude() != 0) {
+                    return false;
+                }
+
+                try {
+                    Thread.sleep(500);
+                } catch (InterruptedException e) {

```

Comment:
This doesn't feel right to me. If we don't think `recorder.getMaxAmplitude()`, lets not use it at all instead of hacking something ambiguous here. I would suggest that we only do a one time check on `recorder.getMaxAmplitude()` and treat it as reliable unless we find otherwise